### ISSUE TITLE
Able to pay an invoice from CLN to Phoenix but not one obtained via an offer

### ISSUE THREAD

**Issue by hMsats** [ACINQ/phoenix#608]
Phoenix Android v2.3.6, see also the corresponding CLN [issue](https://github.com/ElementsProject/lightning/issues/7513)

I'm able to pay an invoice from CLN to my Phoenix wallet  (obtained via Receive in the wallet):

```
lightning-cli pay lnbc1pn27e5cpp56saurku7de8tg2xkmuhj89674s7g4hw5xqzc6newpy6yymq8klfscqpjsp5qkxuqlptp5y62x5gwxclw8my84gey6tv8hxxns626r6jfeun2uas9q7sqqqqqqqqqqqqqqqqqqqsqqqqqysgqdqqmqz9gxqyjw5qrzjqwryaup9lh50kkranzgcdnn2fgvx390wgj5jd07rwr3vxeje0glclluxwv9rqu2e4sqqqqlgqqqqqeqqjqnz40u8l6cqfufhgyf2fajstqs07fstu0u48xf9alxkjmyjkcwdvkkd98mc2xh5j7pu0llxh8fqw0886e9jwkhy2pr6lnddwf7p34uasp74l3p6 41000
{
   "destination": "021d42b28b34c9ff3c98fa8408b4b58e4ecf2acf7fcdc171438e86730a307159ac",
   "payment_hash": "d43bc1db9e6e4eb428d6df2f23975eac3c8addd430058d4f2e0934426c07b7d3",
   "created_at": 1722771243.434156162,
   "parts": 1,
   "amount_msat": 41000,
   "amount_sent_msat": 42004,
   "payment_preimage": "1335b32a36ae30825bcb31af116dd490a87790c11f129bc72507aa2e11516414",
   "status": "complete"
}
```

When I fetch an invoice via an offer (obtained in the Phoenix wallet via "show reusable QR code"):

```
lightning-cli fetchinvoice lno1zrxq8pjw7qjlm68mtp7e3yvxee4y5xrgjhhyf2fxhlphpckrvevh50u0qwa3w0uk0h7vytuuytqemllkhfnaux2f2lwe0k4zhnxeu7gvz7pv2qsraavdg00v4xmpjdm7sah7qrjn06eq9tqk2jr3axlszglfct38sjxqqv7ca9z7qyjxg6jq7t900nswque9uznddry9kksky62v303pr4wzyw3pvaeqwncysfzx7qt7myhv8f4v9pf8qw3aaueuy3sm605k20vnlk5yme8g3aldws9maq29mzp788ex6qhujqqstzv26hlksmcr5efxtplpz7fswg 42000
{
   "invoice": "lni1qqgyq9a6ddqn53yhpacerdn8aze4gyxvqwryaup9lh50kkranzgcdnn2fgvx390wgj5jd07rwr3vxeje0glc7qamzulevl0ucghecgkpnhlldwn8mcv5j47ajld290xdneusc9uzc5pq8m6c6s77e2dkrymhapm0uq89xl4jq2kpv4y8r6dlqy37nshz0pyvqqea3629uqfyv34ypuk27l8qupejtc9x66xgtddpvf55ezlzz82uygazzemjqa8sfqjyduqhakfwcwn2c2zjwqarmmencfrph5lfv57e8ldgfhjw3rm76aqth6q5tkyruw0jd5p0eyqpqkyc440ldphs8fjjvkr7z9unqujjq2jpqkppqfm8fn3u38f0elc73mdrf0ltz90fhjvpvx5pppld8lnu9wtt8ydp8g8aqxvs8pjw7qjlm68mtp7e3yvxee4y5xrgjhhyf2fxhlphpckrvevh50u0qwnlg2qecnkxfdhlu7fmeyprc4vg75ua7tga66nh9ej3ej0af6m7qqsrmm85kfmr0hs4y0zcq0dxa55fq25aetj58nc7hmzss2y4z7e7sz6sq3z6ds3cc4yath7k40g26ye96klg9rhl6x3flskepu7y2t8t9fzuju6a5dkejnqf8j2agpkzsqlu64hf2uz98vcmjuhx09h37vj255t6v42klnqqyj3mmlrunhvvkvzjk4due3zwhtz78gnzfd3qhqxxay3k6x88qfz3qrxqn28wsutdnsxccjgz434n0ggzl9wny88nak6skuw79c3xvacskpq9rlj5tz95sl2nu70s0w7aln3ew7y2me0dunnfn222tck7fqcsxu67j69yqasmkpcsqgghkrq2f9a2w0vmes76wja8raevpgcmcpf3u373pu0gjp29zqz2s3gyh295hvknr43al84kl7989465df9dghzszruh2q2s58vm6t74j0yxlpdl0rms7vmzvvyjpk3evfgpm83gpsue6leaxauh36g6fm0nw452kqen4f6mdj5ajpscvk87eup2uwp8nnlr2y2v9sfuaquv7awrcj3pcqqqq05qqqqqvsqfqqqqqqqqqqqqqyqqqqqqqqq2gyqqqzjqge40v992vqfu4qswff99pupu7anxv98sekkpja3segz3kn898g43qwy80yk5tg885yd2q2jpptsrqgqqpvppqw3aaueuy3sm605k20vnlk5yme8g3aldws9maq29mzp788ex6qhunuzq0pv8c0zch6crg05ccvykz2tp695aylwhdsjuymqf2n0g6x0mxv7ct7exhduu2cpgmxl75lngzvqp3clefraxx28kvgt9qfu2f4xeaxs",
   "changes": {}
}
```

decoded the invoice:

```
lightning-cli decode lni1qqgyq9a6ddqn53yhpacerdn8aze4gyxvqwryaup9lh50kkranzgcdnn2fgvx390wgj5jd07rwr3vxeje0glc7qamzulevl0ucghecgkpnhlldwn8mcv5j47ajld290xdneusc9uzc5pq8m6c6s77e2dkrymhapm0uq89xl4jq2kpv4y8r6dlqy37nshz0pyvqqea3629uqfyv34ypuk27l8qupejtc9x66xgtddpvf55ezlzz82uygazzemjqa8sfqjyduqhakfwcwn2c2zjwqarmmencfrph5lfv57e8ldgfhjw3rm76aqth6q5tkyruw0jd5p0eyqpqkyc440ldphs8fjjvkr7z9unqujjq2jpqkppqfm8fn3u38f0elc73mdrf0ltz90fhjvpvx5pppld8lnu9wtt8ydp8g8aqxvs8pjw7qjlm68mtp7e3yvxee4y5xrgjhhyf2fxhlphpckrvevh50u0qwnlg2qecnkxfdhlu7fmeyprc4vg75ua7tga66nh9ej3ej0af6m7qqsrmm85kfmr0hs4y0zcq0dxa55fq25aetj58nc7hmzss2y4z7e7sz6sq3z6ds3cc4yath7k40g26ye96klg9rhl6x3flskepu7y2t8t9fzuju6a5dkejnqf8j2agpkzsqlu64hf2uz98vcmjuhx09h37vj255t6v42klnqqyj3mmlrunhvvkvzjk4due3zwhtz78gnzfd3qhqxxay3k6x88qfz3qrxqn28wsutdnsxccjgz434n0ggzl9wny88nak6skuw79c3xvacskpq9rlj5tz95sl2nu70s0w7aln3ew7y2me0dunnfn222tck7fqcsxu67j69yqasmkpcsqgghkrq2f9a2w0vmes76wja8raevpgcmcpf3u373pu0gjp29zqz2s3gyh295hvknr43al84kl7989465df9dghzszruh2q2s58vm6t74j0yxlpdl0rms7vmzvvyjpk3evfgpm83gpsue6leaxauh36g6fm0nw452kqen4f6mdj5ajpscvk87eup2uwp8nnlr2y2v9sfuaquv7awrcj3pcqqqq05qqqqqvsqfqqqqqqqqqqqqqyqqqqqqqqq2gyqqqzjqge40v992vqfu4qswff99pupu7anxv98sekkpja3segz3kn898g43qwy80yk5tg885yd2q2jpptsrqgqqpvppqw3aaueuy3sm605k20vnlk5yme8g3aldws9maq29mzp788ex6qhunuzq0pv8c0zch6crg05ccvykz2tp695aylwhdsjuymqf2n0g6x0mxv7ct7exhduu2cpgmxl75lngzvqp3clefraxx28kvgt9qfu2f4xeaxs
{
   "type": "bolt12 invoice",
   "offer_id": "5a42852313c4fed799792e43a63b115507b4044e74eeb5424a76a9f9467c48f9",
   "offer_paths": [
      {
         "first_node_id": "03864ef025fde8fb587d989186ce6a4a186895ee44a926bfc370e2c366597a3f8f",
         "blinding": "03bb173f967dfcc22f9c22c19dfff6ba67de194957dd97daa2bccd9e790c1782c5",
         "path": [
            {
               "blinded_node_id": "03ef58d43deca9b619377e876fe00e537eb202ac1654871e9bf0123e9c2e27848c",
               "encrypted_recipient_data": "d8e945e0124646a40f2caf7ce0e07325e0a6d68c85b5a162694c8be211d5c223a216772074f0482446f017ed92ec3a6ac28527"
            },
            {
               "blinded_node_id": "03a3def33c2461bd3e9653d93fda84de4e88f7ed740bbe8145d883e39f26d02fc9",
               "encrypted_recipient_data": "5898ad5ff686f03a6526587e11793072"
            }
         ]
      }
   ],
   "invreq_metadata": "4017ba6b413a44970f7191b667e8b354",
   "invreq_payer_id": "027674ce3c89d2fcff1e8eda34bfeb115e9bc98161a81087ed3fe7c2b96b391a13",
   "invreq_amount_msat": 42000,
   "invoice_paths": [
      {
         "first_node_id": "03864ef025fde8fb587d989186ce6a4a186895ee44a926bfc370e2c366597a3f8f",
         "blinding": "03a7f42819c4ec64b6ffe793bc9023c5588f539df2d1dd6a772e651cc9fd4eb7e0",
         "payinfo": {
            "fee_base_msat": 1000,
            "fee_proportional_millionths": 100,
            "cltv_expiry_delta": 144,
            "features": ""
         },
         "path": [
            {
               "blinded_node_id": "03decf4b27637de1523c5803da6ed28902a9dcae543cf1ebec508289517b3e80b5",
               "encrypted_recipient_data": "5a6c238c549d5dfd6abd0ad1325d5be828effd1a29fc2d90f3c452ceb2a45c9735da36d994c093c95d406c2803fcd56e9570453b31b972e6796f1f324aa517a65556fcc0"
            },
            {
               "blinded_node_id": "024a3bdfc7c9dd8cb3052b55bccc44ebac5e3a2624b620b80c6e9236d18e702451",
               "encrypted_recipient_data": "09a8ee8716d9c0d8c4902ac6b37a102f95d321cf3edb50b71de2e22667710b04051fe54588b487d53e79f07bbddfce397788ade5ede4e699a94a5e2de483103735e968a40761bb071002117b0c0a497aa73d9bcc3da74ba71f72c0a31bc0531e47d10f1e8905451004a84504ba8b4bb2d31d63df9eb6ff8a72d7546a4ad45c5010f9750150a1d9bd2fd593c86f85bf78f70f3362630920da3962501d9e280c399d7f3d377978e91a4edf37568ab0333aa75b6ca9d90618658fecf02ae38279cfe35114c2c13ce838cf75c3c4"
            }
         ]
      }
   ],
   "invoice_created_at": 1722769738,
   "invoice_relative_expiry": 60,
   "invoice_payment_hash": "e4a4a50f03cf7666614f0cdac197630ca051b4ce53a2b103887792d45a0e7a11",
   "invoice_amount_msat": 42000,
   "features": "020000",
   "invoice_node_id": "03a3def33c2461bd3e9653d93fda84de4e88f7ed740bbe8145d883e39f26d02fc9",
   "signature": "78587c3c58beb0343e98c309612961d169d27dd76c25c26c0954de8d19fb333d85fb26bb79c56028d9bfea7e68130018e3f948fa6328f6621650278a4d4d9e9a",
   "valid": true
}
```

and try to pay the invoice:

```
lightning-cli pay lni1qqgyq9a6ddqn53yhpacerdn8aze4gyxvqwryaup9lh50kkranzgcdnn2fgvx390wgj5jd07rwr3vxeje0glc7qamzulevl0ucghecgkpnhlldwn8mcv5j47ajld290xdneusc9uzc5pq8m6c6s77e2dkrymhapm0uq89xl4jq2kpv4y8r6dlqy37nshz0pyvqqea3629uqfyv34ypuk27l8qupejtc9x66xgtddpvf55ezlzz82uygazzemjqa8sfqjyduqhakfwcwn2c2zjwqarmmencfrph5lfv57e8ldgfhjw3rm76aqth6q5tkyruw0jd5p0eyqpqkyc440ldphs8fjjvkr7z9unqujjq2jpqkppqfm8fn3u38f0elc73mdrf0ltz90fhjvpvx5pppld8lnu9wtt8ydp8g8aqxvs8pjw7qjlm68mtp7e3yvxee4y5xrgjhhyf2fxhlphpckrvevh50u0qwnlg2qecnkxfdhlu7fmeyprc4vg75ua7tga66nh9ej3ej0af6m7qqsrmm85kfmr0hs4y0zcq0dxa55fq25aetj58nc7hmzss2y4z7e7sz6sq3z6ds3cc4yath7k40g26ye96klg9rhl6x3flskepu7y2t8t9fzuju6a5dkejnqf8j2agpkzsqlu64hf2uz98vcmjuhx09h37vj255t6v42klnqqyj3mmlrunhvvkvzjk4due3zwhtz78gnzfd3qhqxxay3k6x88qfz3qrxqn28wsutdnsxccjgz434n0ggzl9wny88nak6skuw79c3xvacskpq9rlj5tz95sl2nu70s0w7aln3ew7y2me0dunnfn222tck7fqcsxu67j69yqasmkpcsqgghkrq2f9a2w0vmes76wja8raevpgcmcpf3u373pu0gjp29zqz2s3gyh295hvknr43al84kl7989465df9dghzszruh2q2s58vm6t74j0yxlpdl0rms7vmzvvyjpk3evfgpm83gpsue6leaxauh36g6fm0nw452kqen4f6mdj5ajpscvk87eup2uwp8nnlr2y2v9sfuaquv7awrcj3pcqqqq05qqqqqvsqfqqqqqqqqqqqqqyqqqqqqqqq2gyqqqzjqge40v992vqfu4qswff99pupu7anxv98sekkpja3segz3kn898g43qwy80yk5tg885yd2q2jpptsrqgqqpvppqw3aaueuy3sm605k20vnlk5yme8g3aldws9maq29mzp788ex6qhunuzq0pv8c0zch6crg05ccvykz2tp695aylwhdsjuymqf2n0g6x0mxv7ct7exhduu2cpgmxl75lngzvqp3clefraxx28kvgt9qfu2f4xeaxs
{
   "code": 203,
   "message": "failed: WIRE_INVALID_ONION_BLINDING (reply from remote)",
   "id": 262,
   "failcode": 49176,
   "failcodename": "WIRE_INVALID_ONION_BLINDING",
   "bolt12": "lni1qqgyq9a6ddqn53yhpacerdn8aze4gyxvqwryaup9lh50kkranzgcdnn2fgvx390wgj5jd07rwr3vxeje0glc7qamzulevl0ucghecgkpnhlldwn8mcv5j47ajld290xdneusc9uzc5pq8m6c6s77e2dkrymhapm0uq89xl4jq2kpv4y8r6dlqy37nshz0pyvqqea3629uqfyv34ypuk27l8qupejtc9x66xgtddpvf55ezlzz82uygazzemjqa8sfqjyduqhakfwcwn2c2zjwqarmmencfrph5lfv57e8ldgfhjw3rm76aqth6q5tkyruw0jd5p0eyqpqkyc440ldphs8fjjvkr7z9unqujjq2jpqkppqfm8fn3u38f0elc73mdrf0ltz90fhjvpvx5pppld8lnu9wtt8ydp8g8aqxvs8pjw7qjlm68mtp7e3yvxee4y5xrgjhhyf2fxhlphpckrvevh50u0qwnlg2qecnkxfdhlu7fmeyprc4vg75ua7tga66nh9ej3ej0af6m7qqsrmm85kfmr0hs4y0zcq0dxa55fq25aetj58nc7hmzss2y4z7e7sz6sq3z6ds3cc4yath7k40g26ye96klg9rhl6x3flskepu7y2t8t9fzuju6a5dkejnqf8j2agpkzsqlu64hf2uz98vcmjuhx09h37vj255t6v42klnqqyj3mmlrunhvvkvzjk4due3zwhtz78gnzfd3qhqxxay3k6x88qfz3qrxqn28wsutdnsxccjgz434n0ggzl9wny88nak6skuw79c3xvacskpq9rlj5tz95sl2nu70s0w7aln3ew7y2me0dunnfn222tck7fqcsxu67j69yqasmkpcsqgghkrq2f9a2w0vmes76wja8raevpgcmcpf3u373pu0gjp29zqz2s3gyh295hvknr43al84kl7989465df9dghzszruh2q2s58vm6t74j0yxlpdl0rms7vmzvvyjpk3evfgpm83gpsue6leaxauh36g6fm0nw452kqen4f6mdj5ajpscvk87eup2uwp8nnlr2y2v9sfuaquv7awrcj3pcqqqq05qqqqqvsqfqqqqqqqqqqqqqyqqqqqqqqq2gyqqqzjqge40v992vqfu4qswff99pupu7anxv98sekkpja3segz3kn898g43qwy80yk5tg885yd2q2jpptsrqgqqpvppqw3aaueuy3sm605k20vnlk5yme8g3aldws9maq29mzp788ex6qhunuzq0pv8c0zch6crg05ccvykz2tp695aylwhdsjuymqf2n0g6x0mxv7ct7exhduu2cpgmxl75lngzvqp3clefraxx28kvgt9qfu2f4xeaxs",
   "raw_message": "c018deabbd37b638f22278355e7f0ff082ed1f2d52ce86414a243e95843af693b22d",
   "created_at": 1722769748,
   "destination": "03a3def33c2461bd3e9653d93fda84de4e88f7ed740bbe8145d883e39f26d02fc9",
   "payment_hash": "e4a4a50f03cf7666614f0cdac197630ca051b4ce53a2b103887792d45a0e7a11",
   "status": "failed",
   "amount_msat": 43004,
   "amount_sent_msat": 0,
   "erring_index": 2,
   "erring_node": "03864ef025fde8fb587d989186ce6a4a186895ee44a926bfc370e2c366597a3f8f"
}
```

I get `WIRE_INVALID_ONION_BLINDING` and it doesn't work. Is this still normal at this stage or should the payment go through?

**Comment by hMsats** [comment]
I don't know if this is useful but my Phoenix log shows the following when I try to pay an invoice (with CLN) obtained via `fetchinvoice`:

```
2024-08-04 13:34:35,781 INFO  f.a.lightning.io.Peer  - sending OnionMessage(blindingKey=0301f44439abfd987f6dbd75a8bd99ea4f10f47152411be04a70b06c9a2854f789, onionRoutingPacket=OnionRoutingPacket(version=0, publicKey=03ca76055dfed0bca619f2f616c80a9c09a7d43140ea115ced924a28a429f42987, payload=<0 bytes>, hmac=f9ae76e12c936a94b3360f4466d0800feb51ab11c1d3cb197455cfa5b6250b1f)) (truncated payload)
```

all the time I see `payload=<0 bytes>`

**Comment by hMsats** [comment]
Solved by `Phoenix v2.3.7` using the latest `lightning-kmp`! Was able to send 42 sats (plus fees) from my Core Lightning node to my Phoenix wallet and it worked flawlessly:

{
   "destination": "03a3def33c2461bd3e9653d93fda84de4e88f7ed740bbe8145d883e39f26d02fc9",
   "payment_hash": "9706499a74131b690fbdedba534daaa5741b71c2c7c0f088796dcbe5db21ff8f",
   "created_at": 1725198096.980676748,
   "parts": 1,
   "amount_msat": 43004,
   "amount_sent_msat": 43004,
   "payment_preimage": "ccf3b4921b79a2a151ee7358a4584805fe1ed106ff21a0fbec4f1359c0895059",
   "status": "complete"
}

Thanks to all developers for their hard work!
